pkg/cmd/server/api/v1/types.go

```diff
@@ -693,6 +693,9 @@ type EtcdConfig struct {
 type KubernetesMasterConfig struct {
 	// APILevels is a list of API levels that should be enabled on startup: v1beta3 and v1 as examples
 	APILevels []string `json:"apiLevels"`
+	// APIGroupVersionBlacklist is a map of groups to the versions (or *) that should be disabled.
+	APIGroupVersionBlacklist map[string][]string `json:"apiGroupVersionBlacklist"`
```

Comment:
@smarterclayton comments about the approach in the config.  Everything on by default.  Easy to disable an entire group.  Easy to disable some versions of a group.

It needs a little plumbing upstream.  Looks like they break if there's more than one version for the group.